### ISSUE TITLE
Feature Request: Add support for filtering span attributes in net/http package

### ISSUE THREAD

**Issue by BlakeWilliams** [open-telemetry/opentelemetry-go-contrib#837]
We're writing an application and would like the ability to filter the `http.url` attribute that is automatically set in spans. It's possible the url can contain secrets or other information that we would like to avoid sending to third parties.

I'd be happy to spend time getting this working and open a PR if there's interest in adding this functionality to the `net/http` package.

Definitely open to thoughts on what the API should look like, but maybe it could be an option on the [`config` struct](https://github.com/open-telemetry/opentelemetry-go-contrib/blob/main/instrumentation/net/http/otelhttp/config.go#L34)?

I'm imagining something roughly like:

```go
transport := otelhttp.NewTransport(http.DefaultTransport, WithSpanAttributeFilter(attributeFilter))

func attributeFilter(kv attribute.KeyValue, r *http.Request) bool {
    return false;
}
```

**Comment by Aneurysm9** [comment]
Would this be better as a [`SpanProcessor`](https://pkg.go.dev/go.opentelemetry.io/otel/sdk@v1.0.0-RC1/trace#SpanProcessor) on the `TracerProvider`?  That would give it access to all spans, not just the ones created by this instrumentation.  Not necessarily opposed to the idea, just wondering if there is a higher-leverage implementation point.

**Comment by BlakeWilliams** [comment]
@Aneurysm9 That's a good question. I took a look at using `SpanProcessor` but couldn't see how it could be composed with the `batchSpanProcessor` that we're currently using.

**Comment by Aneurysm9** [comment]
[Here is an example](https://github.com/open-telemetry/opentelemetry-go/blob/main/sdk/trace/span_processor_filter_example_test.go) showing a filter setup.  The BSP would be provided to the filter processor as a next step.  I'm not sure it would work for this use, though, since `OnEnd` receives a `ReadOnlySpan`.  It could filter out spans entirely, but not remove certain attributes without some gymnastics to reconstruct a writable span.

**Comment by BlakeWilliams** [comment]
> I'm not sure it would work for this use, though, since `OnEnd` receives a `ReadOnlySpan`. It could filter out spans entirely, but not remove certain attributes without some gymnastics to reconstruct a writable span.

Yeah, that's what I was thinking as well. I don't want to filter out the entire span, just some of the attributes.
